libraries/standard/http/utest/http_utest.c

```diff
@@ -201,123 +199,123 @@ static void setupRequestInfo( HTTPRequestInfo_t * pRequestInfo )
+                              HTTP_CONNECTION_CLOSE_VALUE_LEN;
+    setupBuffer( &requestHeaders );
 
     /* Happy Path testing. */
-    expectedHeaderLen = HTTP_TEST_PREFIX_HEADER_LEN +
-                        HTTP_CONNECTION_CLOSE_VALUE_LEN;
-    numBytes = snprintf( expectedHeader, expectedHeaderLen + 1,
+    numBytes = snprintf( ( char * ) expectedHeaders.buffer, sizeof( expectedHeaders.buffer ),
                          HTTP_TEST_HEADER_FORMAT,
                          HTTP_METHOD_GET, HTTP_TEST_REQUEST_PATH,
                          HTTP_PROTOCOL_VERSION,
                          HTTP_USER_AGENT_FIELD, HTTP_USER_AGENT_VALUE,
                          HTTP_HOST_FIELD, HTTP_TEST_HOST_VALUE,
                          HTTP_CONNECTION_FIELD, HTTP_CONNECTION_CLOSE_VALUE );
-    TEST_ASSERT_EQUAL( numBytes, expectedHeaderLen );
-    test_err = HTTPClient_InitializeRequestHeaders( &requestHeaders, &requestInfo );
-    TEST_ASSERT_EQUAL_MEMORY( requestHeaders.pBuffer, expectedHeader, expectedHeaderLen );
-    TEST_ASSERT_EQUAL( requestHeaders.headersLen, expectedHeaderLen );
-    TEST_ASSERT_EQUAL( test_err, HTTP_SUCCESS );
+    /* Make sure that the entire pre-existing data was printed to the buffer. */
+    TEST_ASSERT_GREATER_THAN( 0, numBytes );
+    TEST_ASSERT_LESS_THAN( sizeof( expectedHeaders.buffer ), ( size_t ) numBytes );
+
+    httpStatus = HTTPClient_InitializeRequestHeaders( &requestHeaders, &requestInfo );
+    TEST_ASSERT_EQUAL( HTTP_SUCCESS, httpStatus );
+    TEST_ASSERT_EQUAL( expectedHeaders.dataLen, requestHeaders.headersLen );
+    TEST_ASSERT_EQUAL_MEMORY( expectedHeaders.buffer, requestHeaders.pBuffer,
+                              expectedHeaders.dataLen );
 }
 
 /**
  * @brief Test NULL parameters, following order of else-if blocks in the HTTP library.
  */
-void test_Http_InitializeRequestHeaders_invalid_params()
+void test_Http_InitializeRequestHeaders_Invalid_params()
 {
-    HTTPStatus_t test_err = HTTP_INTERNAL_ERROR;
+    HTTPStatus_t httpStatus = HTTP_INTERNAL_ERROR;
     HTTPRequestHeaders_t requestHeaders = { 0 };
     HTTPRequestInfo_t requestInfo = { 0 };
-    size_t expectedHeaderLen = HTTP_TEST_MAX_INITIALIZED_HEADER_LEN;
-    char expectedHeader[ HTTP_TEST_INITIALIZED_HEADER_BUFFER_LEN ] = { 0 };
 
     /* Test NULL parameters, following order of else-if blocks. */
-    test_err = HTTPClient_InitializeRequestHeaders( NULL, &requestInfo );
-    TEST_ASSERT_EQUAL( test_err, HTTP_INVALID_PARAMETER );
+    httpStatus = HTTPClient_InitializeRequestHeaders( NULL, &requestInfo );
+    TEST_ASSERT_EQUAL( HTTP_INVALID_PARAMETER, httpStatus );
     /* TEST requestInfo.pBuffer == NULL */
-    test_err = HTTPClient_InitializeRequestHeaders( &requestHeaders, &requestInfo );
-    TEST_ASSERT_EQUAL( test_err, HTTP_INVALID_PARAMETER );
-    requestHeaders.pBuffer = httpBuffer;
+    httpStatus = HTTPClient_InitializeRequestHeaders( &requestHeaders, &requestInfo );
```

Comment:
This code is hard to read. Helpful would be ensuring that each test case is accompanied by a comment and that the test cases are separated by an empty line.